finagle-thrift/src/main/scala/com/twitter/finagle/rich.scala

```diff
@@ -447,6 +473,43 @@ trait ThriftRichClient { self: Client[ThriftClientRequest, Array[Byte]] =>
   def newMethodIface[ServiceIface, FutureIface](serviceIface: ServiceIface)(
     implicit builder: MethodIfaceBuilder[ServiceIface, FutureIface]
   ): FutureIface = builder.newMethodIface(serviceIface)
+
+  def multiplex[T](dest: Name, label: String)(build: MultiplexedThriftClient => T): T = {
+    build(new MultiplexedThriftClient(dest, label))
+  }
+
+  def multiplex[T](dest: String, label: String)(build: MultiplexedThriftClient => T): T = {
+    multiplex(Resolver.eval(dest), label)(build)
+  }
+
+  def multiplex[T](dest: String)(build: MultiplexedThriftClient => T): T = {
```

Comment:
mind omitting this and forcing these clients to specify a label? its for their own good ;)